Plasma cutter interaction

ghostsheet opened the pull request:
🆑
:rscadd: Plasma cutters can now through the following: normal walls, floor platings, girders, catwalks and lattices.
:tweak: Handheld drills will now have a delay and sound when drilling through girders.
/🆑

Make the plasma cutter a universal cutting tool.
Allow them to cut through:
Catwalks
Lattices
Girders
Floors
Walls
Windows
Low walls
(reinforce walls are untouched although plasma cutter can already be used on them in place of the welder steps)

Also fixed up some messy mounted plasma cutter code (ty Psi)
Also added sound back for the drilling through girders
Fixed up a bit more code with the girders and floor code (ty Psi again)

comma committed:
New exoplanet ruin - EC wreck

Ancient STL EC ship crashlanded

comma committed:
Species organ tweaks

Adds ability to partally override internal organs list, there as similar for external.
Gets rid of bioprint products list, since it's been used for 1 (one) species, and is /already/ unmaintained.
Instead organ decides whether it can be printed or not, and bioprinter just gets all printable organs from species.

comma committed:
Moves eye icon generation into eye icon from head

ghostsheet committed:
Ore Unloader Tweaks and More!

Hubblenaut committed:
Fixes the chameleon items and includes some new

- Fixes chameleon items having missing items.
- Fixes chameleon items not assuming the original item's color.
- Makes a chameleon headset and includes it in the kit.
- Makes a chameleon accessory and includes three in the kit.
- The list of available items to copy will be sorted by name now.
- Replaces New() with Initialize().

WatermelonsEverywhere committed:
Changes the advanced mop's water synthesizer to synthesize space cleaner

Commit:
Make "Anomaly power battery" usable

Fixes an ancient runtime error which results in Anomaly power batteries becoming unusable when inserted into an Anomaly power utilizer (adds a null check on line 56, and some null assignments).

Also fixes an oversight which causes the battery charge to instantly drop from 300 to 0 upon activation.

SierraKomodo committed:
Fix deck 1 checkpoint shutters buttons
Properly fixes them this time.

AtomicWorm committed:
Adds gendered icons for a few more clothing items

Commit:
Makes "Anomaly power battery" usable

Implements suggested changes

Higgin committed:
Adds slowdown to tactical plates, changes desc.

PsiOmegaDelta commented on code/game/objects/structures/girders.dm:
```diff
@@ -78,15 +78,19 @@
-		dismantle()
+		playsound(src.loc, 'sound/weapons/Genhit.ogg', 100, 1)
+		if(do_after(user,40,src))
+			if(!src) return
```
`if(QDELETED(src)) return`

PsiOmegaDelta commented on code/game/turfs/simulated/floor_attackby.dm:
```diff
@@ -128,6 +128,15 @@
+		else if(istype(C, /obj/item/weapon/gun/energy/plasmacutter) && (is_plating()) && !broken && !burnt)
+			playsound(src, 'sound/items/Welder.ogg', 80, 1)
+			visible_message("<span class='notice'>[user] has started melting the plating's reinforcements!</span>")
+			if(do_after(user, 3 SECONDS))
```
Need to re-validate the checks above (`is_plating()`, etc.) after a `do_after()`

ghostsheet committed:
Plasma Cutter interaction

ghostsheet committed:
Plasma Cutter interaction

mikomyazaki committed:
Adds a bottle of universal enzyme to the cooking supply pack.

PsiOmegaDelta committed:
Remove emag copy-paste, cleanup chameleon code

The chameleon /disguise() proc now properly qdels temporary items
generate_chameleon_choices() handles receiving either a list or type
Emag now re-uses existing chameleon code
Removes redundant update_icon() call

comma committed:
Gives barren planets proper area

With base turf

Bxil committed:
Main map is now a valid BSA target

PsiOmegaDelta committed:
Automatic changelog update [ci skip]

Hubblenaut committed:
Fixes window icons and material count

Flying-loulou committed:
Adds the EMS helmet

Flying-loulou committed:
Adds the EMT Chest Rig

Flying-loulou committed:
Adds the EMS helmet and EMT Chest Rig in the medical technician's locker.

Flying-loulou committed:
Removes the labcoat from the MT locker.

Higgin committed:
changes times to a define

babydoll committed:
Merge pull request #25338 from WatermelonsEverywhere/augmentedsanitation

Changes the advanced mop's water synthesizer to synthesize space cleaner

PsiOmegaDelta committed:
Automatic changelog generation for PR #25338 [ci skip]

babydoll committed:
Merge pull request #25354 from comma/fixies

Fixies

PsiOmegaDelta committed:
Automatic changelog generation for PR #25344 [ci skip]

comma committed:
Spacesuit resprite

Sprites by WalterJe
Superiour fishbowl aesthetic is upon us.
Made fishbowl helmet a separate item because it has toggleable tint cause I couldn't bring myself to sacrifice transparent bowl for sake of keeping them facehiding/flashresistant like normal helmets, and also because it has different light overlay and I didn't want to test every single space helmet in case it doesn't override it.

Replaced non-bowl helmets in all spots I found them used.

afterthought2 commented on code/game/objects/structures/catwalk.dm:
```diff
@@ -77,6 +77,18 @@
+		if(plated_tile)
+			new plated_tile.build_type(src.loc)
+		qdel(src)
+		return
```
I am detecting egregious copypaste of the block above this. Pull out into a shared proc.

afterthought2 commented on code/game/turfs/simulated/floor_attackby.dm:
```diff
@@ -128,6 +130,16 @@
+				playsound(src, 'sound/items/Welder.ogg', 80, 1)
+				burnt = 1
+				remove_decals()
+				update_icon()
```
Further copypaste detected here.

afterthought2 reviewed (changes requested):
If you want to have your thing do something that is already done by something else, the correct way to do this is not to copypaste that code block, but rather to pull out the shared code into a proc and call that proc in both places. That helps make the code more maintainable.

afterthought2 committed:
Merge pull request #25350 from mikomyazaki/enzyme_supplypack

Adds a bottle of universal enzyme to the cooking supply pack.

PsiOmegaDelta committed:
Automatic changelog generation for PR #25350 [ci skip]

PsiOmegaDelta committed:
Merge pull request #25360 from Flying-loulou/Medical-Technicians-equipment

New Medical technicians equipments

PsiOmegaDelta committed:
Automatic changelog generation for PR #25360 [ci skip]

comma committed:
Unapolegetic ballistc inflatables nerf

Drastically lowered health of inflatables so they can't tank stuff.
Made bullets pass through them if they destroy it.
Fixed bullets failing to keep going if thing they passed through was qdel'd in proccess (they were put in nullspace with the thing)

comma committed:
Tightened up printability checks

Now checking few more things aside from straight up NO PRINT THIS flag - vital organs, non-replacible stuff, robotic-from-start stuff.

PsiOmegaDelta committed:
Automatic changelog generation for PR #25365 [ci skip]

mattparker124 committed:
Green+Black Tea Descriptions

Added missing green tea descriptions (they previously got described as black tea when looked at) and edited black tea descriptions to clarify the type

comma committed:
Tweaks some surgery steps to reduce popups again

Amputation now doesn't pop up as a choice if there's an incision. You couldn't do it anyway if there was one, it was a dummy choice.

When doing necrosis surgery, skip choice box if there's only one organ to pick from.

Sterilize step now doesn't pop up as  a choice if your tool isn't holding right chems.

afterthought2 committed:
Merge pull request #25380 from Ostracism/Ostracism-patch-1

Green+Black Tea Descriptions

babydoll committed:
Merge pull request #25376 from comma/poppop

Unapolegetic ballistc inflatables nerf

babydoll committed:
Merge pull request #25353 from comma/syrette

Syringe resprite

babydoll committed:
Merge pull request #25346 from AtomicWorm/boobs2.0

Adds gendered icons to a few more clothing items.

babydoll committed:
Merge pull request #25351 from Baystation12/PurgeTheUnclean

Remove emag copy-paste, cleanup chameleon code

Spookerton committed:
narrates can be styled

:cl:
tweak: Admin narrations can have text styling.
/:cl:

Higgin committed:
Kills spawn, uses timer instead.

comma committed:
De-grillifies most windows

Went around and replaced most windows with no grille version.
Setup is like this:
Do we need to stop people from breaking this window / it's space facing? - reinforced and grille
Does this window need to withstand some nastiness fairly often (e.g hangar or windows with shutters e.g. sci or med quarantine)? - reinforced.
Otherwise it's simple window.

Did couple small changes along the way:
-Moved stool in exploration prep to paperdesk/computers instead of having it next to racks
-Removed weird window in supply office that offered view of 1 tile maint corridor.
-Moved noticeboard at Mess doors couple tiles to the right, to put a window between doors instead (for putting people through in bar fights)

PsiOmegaDelta committed:
Automatic changelog generation for PR #25385 [ci skip]

comma committed:
Merge pull request #25355 from RyanSmake/bsarea

Fixes shooting at the Torch with the BSA

PsiOmegaDelta committed:
Automatic changelog generation for PR #25355 [ci skip]

comma committed:
Merge pull request #25347 from Higgin/tacticruel-fate

Buffs tactical armor brute/bullet values.

PsiOmegaDelta committed:
Automatic changelog generation for PR #25347 [ci skip]

comma committed:
Merge pull request #25342 from ChebPomidorov/patch-1

Makes "Anomaly power battery" usable

comma committed:
Merge pull request #25387 from Higgin/pls-respond

Adminhelps not taken in five minutes automatically close.

PsiOmegaDelta committed:
Automatic changelog generation for PR #25387 [ci skip]

comma committed:
Robustifies lost in space checks for atoms

comma committed:
Fixes #25246

Removes some chaokocode that claimed to clean up trajectory segments but actually flat out deleted bullets in space. Not sure how it wasn't noticed earlier, any shot originating in space would've been deleted right away.

comma committed:
Fixes #20483

Jckets were not showing up cause they had same name

comma committed:
Adds clue sounds

They play when you examine something with forensic traces on it while in noir mode.

comma committed:
Makes airlock do warning sounds when starting cycling

Uses same sound as disposal chute

xales committed:
Merge pull request #25395 from comma/noir

Adds clue sounds

xales committed:
Merge pull request #25394 from comma/fuckyouleatherman

Fixes #20483

xales committed:
Merge pull request #25393 from comma/pewpew

Fixes #25246

xales committed:
Merge pull request #25388 from comma/tfwnogrill

De-grillifies most windows

xales committed:
Merge pull request #25382 from comma/killboxes

Tweaks some surgery steps to reduce popups again

xales committed:
Merge pull request #25392 from comma/spacemech

Robustifies lost in space checks for atoms

ghostsheet committed:
Plasma cutter interaction fixes

As afterthough requested, I cleaned up my code
Add window interaction
Add low wall interaction
Fixed up a bit mounted module buggery

ghostsheet committed:
Plasma cutter interaction fixes

As afterthough requested, I cleaned up my code
Add window interaction
Add low wall interaction
Fixed up a bit mounted module buggery

ghostsheet commented:
Don't merge yet, need to add eyecheck proc to the plasma cutter

PsiOmegaDelta committed:
Merge pull request #25397 from comma/wahwah

Makes airlock do warning sounds when starting cycling

PurpleMartinJCK committed:
Merge pull request #25237 from comma/eyessss

Species organ tweaks

comma committed:
Merge pull request #25324 from Hubblenaut/chameleon

Fixes the chameleon items and includes some new

PsiOmegaDelta committed:
Automatic changelog generation for PR #25324 [ci skip]

ChaosAlpha committed:
Fix robots click-dragging on catwalks

babydoll committed:
Merge pull request #25374 from comma/glubglub

Default and Emergency Spacesuit resprite

babydoll committed:
Merge pull request #25386 from comma/ecwreck

New exoplanet ruin - EC wreck

comma committed:
Merge pull request #25406 from ChaosAlpha/robo-catwalk-drag

Fix robots click-dragging on catwalks

PsiOmegaDelta committed:
Automatic changelog generation for PR #25406 [ci skip]

afterthought2 committed:
Merge pull request #25356 from Hubblenaut/windows

Fixes window icons and material count

afterthought2 commented:
Conflicts.

ghostsheet committed:
Merge branch 'Plasma-Cutter-Buff' of https://github.com/ghostsheet/Baystation12 into Plasma-Cutter-Buff

ghostsheet commented:
uh... oh....

ghostsheet commented:
not.... good

comma committed:
Syringe resprite

comma committed:
Runtime fixes

Fixes #25313
Fixes #25339
